File: backend/app/schemas/messenger.py
"""Pydantic schemas for Messenger webhook responses and CRM read API."""

from __future__ import annotations

from datetime import datetime

from pydantic import BaseModel, ConfigDict, Field


class WebhookAcceptedResponse(BaseModel):
    received: bool = True
    events_processed: int


class PaginationMeta(BaseModel):
    total: int
    page: int
    page_size: int
    has_next: bool
    has_prev: bool


class ConversationResponse(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: str
    page_id: str
    psid: str
    customer_name: str | None
    customer_avatar_url: str | None = None
    last_message_at: datetime | None
    last_message_preview: str | None = None
    unread_count: int = 0
    created_at: datetime
    updated_at: datetime


class ConversationListResponse(BaseModel):
    items: list[ConversationResponse]
    meta: PaginationMeta


class MessageResponse(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: str
    conversation_id: str
    sender_psid: str | None = None
    recipient_page_id: str
    mid: str
    event_type: str
    is_from_page: bool
    text: str | None
    attachments: list[dict] | None = None
    postback_payload: str | None
    fb_timestamp_ms: int | None
    sent_at: datetime | None
    created_at: datetime


class MessageListResponse(BaseModel):
    items: list[MessageResponse]
    meta: PaginationMeta


class MarkConversationReadResponse(BaseModel):
    conversation_id: str
    last_read_at: datetime | None
    unread_count: int
    already_read: bool


class SendMessageRequest(BaseModel):
    text: str = Field(min_length=1, max_length=1000)

File: desktop/src/types/messenger.ts
export type PaginationMeta = {
  total: number;
  page: number;
  page_size: number;
  has_next: boolean;
  has_prev: boolean;
};

export type Conversation = {
  id: string;
  page_id: string;
  psid: string;
  customer_name: string | null;
  customer_avatar_url: string | null;
  last_message_at: string | null;
  last_message_preview: string | null;
  unread_count: number;
  created_at: string;
  updated_at: string;
};

export type ConversationListResponse = {
  items: Conversation[];
  meta: PaginationMeta;
};

export type Message = {
  id: string;
  conversation_id: string;
  sender_psid: string | null;
  recipient_page_id: string;
  mid: string;
  event_type: string;
  is_from_page: boolean;
  text: string | null;
  attachments: unknown[] | null;
  postback_payload: string | null;
  fb_timestamp_ms: number | null;
  sent_at: string | null;
  created_at: string;
};

export type MessageListResponse = {
  items: Message[];
  meta: PaginationMeta;
};

export type MarkConversationReadResponse = {
  conversation_id: string;
  last_read_at: string | null;
  unread_count: number;
  already_read: boolean;
};

export type SendMessageRequest = {
  text: string;
};
